### PR TITLE
Add unique_name() to Input. Use it to generate filename in OnDiskCorpus

### PR DESCRIPTION
**Pull request by s1341**

### CODE REVIEW

**Comment by domenukk** [comment]
The testcase fails, it looks like we depend on multiple versions of ahash?

**Comment by s1341** [comment]
I'll fix it tomorrow.

File: libafl/src/corpus/ondisk.rs [hunk 1]
```diff
@@ -50,7 +50,9 @@ where
     fn add(&mut self, mut testcase: Testcase<I>) -> Result<usize, Error> {
         if testcase.filename().is_none() {
             // TODO walk entry metadata to ask for pices of filename (e.g. :havoc in AFL)
-            let filename = self.dir_path.join(format!("id_{}", &self.entries.len()));
+            let filename = self
+                .dir_path
+                .join(testcase.input().as_ref().unwrap().unique_name());
```
Comment (andreafioraldi) [inline libafl/src/corpus/ondisk.rs:55]: maybe `&self.entries.len()` can be a parameter passed to the method

File: libafl/src/inputs/bytes.rs [hunk 2]
```diff
@@ -48,6 +51,13 @@ impl Input for BytesInput {
         file.read_to_end(&mut bytes)?;
         Ok(BytesInput::new(bytes))
     }
+
+    /// Retrieve a unique name for this input
+    fn unique_name(&self) -> String {
```
Comment (andreafioraldi) [inline libafl/src/inputs/bytes.rs:56]: I'd say, maybe you should rename it `generate_name(idx: usize)` as we don't want necessary to ensure that is unique

**Review by andreafioraldi — CHANGES_REQUESTED** [review]

File: libafl/src/inputs/bytes.rs (hunk 2 above)
Comment (s1341) [reply libafl/src/inputs/bytes.rs:56]: where `idx` is the `self.entries.len()`?

File: libafl/src/corpus/ondisk.rs (hunk 1 above)
Comment (s1341) [reply libafl/src/corpus/ondisk.rs]: Done.

File: libafl/src/inputs/bytes.rs (hunk 2 above)
Comment (s1341) [reply libafl/src/inputs/bytes.rs]: Done.

**Comment by domenukk** [comment]
@andreafioraldi are we fine with this tiny change to break versioning? Alternatively, we could provide a default impl now. Or upgrade version number

**Comment by andreafioraldi** [comment]
> @andreafioraldi are we fine with this tiny change to break versioning? Alternatively, we could provide a default impl now. Or upgrade version number

fino IMO

**Pull request merged**